opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/PlatformResources.java

```diff
@@ -33,113 +37,469 @@
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
+import decodes.db.DatabaseException;
+import decodes.db.DatabaseIO;
+import decodes.db.Platform;
+import decodes.db.PlatformConfig;
+import decodes.db.PlatformList;
+import decodes.db.PlatformSensor;
+import decodes.db.PlatformStatus;
+import decodes.db.RoutingSpec;
+import decodes.db.ScheduleEntry;
+import decodes.db.Site;
+import decodes.db.TransportMedium;
+import decodes.db.ValueNotFoundException;
+import decodes.sql.DbKey;
+import decodes.tsdb.DbIoException;
+import opendcs.dai.PlatformStatusDAI;
+import opendcs.dai.ScheduleEntryDAI;
 import org.opendcs.odcsapi.beans.ApiPlatform;
 import org.opendcs.odcsapi.beans.ApiPlatformRef;
-import org.opendcs.odcsapi.dao.ApiPlatformDAO;
+import org.opendcs.odcsapi.beans.ApiPlatformSensor;
+import org.opendcs.odcsapi.beans.ApiPlatformStatus;
+import org.opendcs.odcsapi.beans.ApiTransportMedium;
 import org.opendcs.odcsapi.dao.DbException;
-import org.opendcs.odcsapi.errorhandling.ErrorCodes;
+import org.opendcs.odcsapi.errorhandling.DatabaseItemNotFoundException;
+import org.opendcs.odcsapi.errorhandling.MissingParameterException;
 import org.opendcs.odcsapi.errorhandling.WebAppException;
-import org.opendcs.odcsapi.hydrojson.DbInterface;
 import org.opendcs.odcsapi.sec.AuthorizationCheck;
-import org.opendcs.odcsapi.util.ApiConstants;
-import org.opendcs.odcsapi.util.ApiHttpUtil;
 
 @Path("/")
-public class PlatformResources
+public class PlatformResources extends OpenDcsResource
 {
 	@Context HttpHeaders httpHeaders;
 
 	@GET
 	@Path("platformrefs")
 	@Produces(MediaType.APPLICATION_JSON)
 	@RolesAllowed({AuthorizationCheck.ODCS_API_GUEST})
-	public Response gePlatformRefs(@QueryParam("tmtype") String tmtype)
-		throws DbException, SQLException
+	public Response getPlatformRefs(@QueryParam("tmtype") String tmtype)
+			throws DbException
 	{
-		HashMap<String, ApiPlatformRef> ret = new HashMap<>();
-		Logger.getLogger(ApiConstants.loggerName).fine("getPlatforms, tmtype=" + tmtype);
-		try (DbInterface dbi = new DbInterface();
-			ApiPlatformDAO platformDAO = new ApiPlatformDAO(dbi))
+		DatabaseIO dbIo = null;
+		Map<String, ApiPlatformRef> ret = new LinkedHashMap<>();
+		try
 		{
-			ArrayList<ApiPlatformRef> platSpecs = platformDAO.getPlatformRefs(tmtype);
+			dbIo = getLegacyDatabase();
+			PlatformList platformList = new PlatformList();
+			platformList = dbIo.readPlatformList(platformList, tmtype);
+			List<ApiPlatformRef> platSpecs = map(platformList);
 			for(ApiPlatformRef ps : platSpecs)
+			{
 				ret.put(ps.getName(), ps);
+			}
+			return Response.status(HttpServletResponse.SC_OK).entity(ret).build();
+		}
+		catch (DatabaseException ex)
+		{
+			throw new DbException("Unable to retrieve platform list", ex);
+		}
+		finally
+		{
+			if (dbIo != null)
```

Comment:
Can't you move `getLegacyDatabase()` out of the try/catch so then you don't need a follow-up null check?